### PR TITLE
refactor(snapshots): Rename, refactor, and re-organize some stuff

### PR DESCRIPTION
**Pull request by InKryption**
Rename, refactor, and re-organize some stuff

Refactor & re-organize more snapshot type stuff

Remove unused `StakeStateV2`

### CODE REVIEW

**Comment by InKryption** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Syndica/sig/440?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#443** <a href="https://app.graphite.dev/github/pr/Syndica/sig/443?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#442** <a href="https://app.graphite.dev/github/pr/Syndica/sig/442?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#441** <a href="https://app.graphite.dev/github/pr/Syndica/sig/441?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#440** <a href="https://app.graphite.dev/github/pr/Syndica/sig/440?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Syndica/sig/440?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#436** <a href="https://app.graphite.dev/github/pr/Syndica/sig/436?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#435** <a href="https://app.graphite.dev/github/pr/Syndica/sig/435?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#434** <a href="https://app.graphite.dev/github/pr/Syndica/sig/434?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/accountsdb/snapshots.zig [hunk 1]
```diff
@@ -1452,22 +1465,22 @@ pub const AccountsDbFields = struct {
...
     bank_fields: BankFields,
     accounts_db_fields: AccountsDbFields,
     /// incremental snapshot fields.
     bank_extra: ExtraFields,
 
-    pub fn deinit(self: SnapshotFields, allocator: std.mem.Allocator) void {
-        self.bank_fields.deinit(allocator);
-        self.accounts_db_fields.deinit(allocator);
-        self.bank_extra.deinit(allocator);
+    pub fn deinit(man: Manifest, allocator: std.mem.Allocator) void {
```
Comment (0xNineteen) [inline src/accountsdb/snapshots.zig:1474]: lets keep consistent with the self pattern

**Review by 0xNineteen — COMMENTED** [review]
small comment then lgtm

File: src/accountsdb/snapshots.zig (hunk 1 above)
Comment (InKryption) [reply src/accountsdb/snapshots.zig]: 6d69637973392273c88393596149d8558455e0d1

**Pull request merged**